go-controller/pkg/ovn/egressgw.go

```diff
@@ -0,0 +1,254 @@
+package ovn
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/klog"
+	"net"
+	"strings"
+)
+
+// addPodExternalGW handles detecting if a pod is serving as an external gateway for namespace(s) and adding routes
+// to all pods in that namespace
+func (oc *Controller) addPodExternalGW(pod *kapi.Pod) error {
+	type Network struct {
+		Name string
+		Ips  []string
+	}
+	routingNamespaceAnnotation := pod.Annotations[routingNamespaceAnnotation]
```

Comment:
deleted my previous comment, the map returns the empty value if it does not exist :)

Reply:
however, we should check if the map exist or it will panic
```go
if pod.Annotations == ni {
return fmt.Errorf(""External gateway pod: %s does not have annotations", pod.Name)
}
```

Reply:
nope, you can read from a `nil` map. you just always get back the empty value

Reply:
wow, I was thinking that if the Annotation field was nil it will panic, thanks for the lesson

Reply:
yeah I was surprised by this too in golang when I started :)